app/grandchallenge/components/tasks.py

```diff
@@ -961,6 +963,51 @@ def add_images_to_component_interface_value(
     civ.image.update_viewer_groups_permissions()
 
 
+@shared_task
+def add_file_to_component_interface_value(
+    *,
+    component_interface_value_pk,
+    user_upload_pk,
+    target_pk,
+    target_app,
+    target_model,
+):
+    user_upload = UserUpload.objects.get(pk=user_upload_pk)
+    civ = get_model_instance(
+        pk=component_interface_value_pk,
+        app_label="components",
+        model_name="componentinterfacevalue",
+    )
+    target = get_model_instance(
+        pk=target_pk,
+        app_label=target_app,
+        model_name=target_model,
+    )
+    error = None
+    with transaction.atomic():
+        try:
+            civ.validate_user_upload(user_upload)
+            civ.full_clean()
+        except ValidationError as e:
+            transaction.set_rollback(True)
+            error = str(e)
```

Comment:
Maybe I'm misunderstanding this, but if this fails, the civ remains in the database without a value or file attached, because it was already created and saved in the previous step, right? Or what does the rollback do in this case? Is that what we want?

Reply:
This is the way James suggested to handle this in https://github.com/comic/grand-challenge.org/pull/2575, so I went with that here as well.

Reply:
You mean in the `add_file_to_display_set`? But there you create the CIV within the atomic block, so that's different, right? Or do you mean a different place?

Reply:
Ah you're right!